testutils/inject/plan.go

```diff
@@ -0,0 +1,25 @@
+package inject
+
+import "go.viam.com/rdk/motionplan"
+
+// Plan is a struct to allow mocking of the motionplan.Plan interface.
+type Plan struct {
+	PathFunc       func() motionplan.Path
```

Comment:
Note: Similar injects import the object itself (in this case motionplan.Plan) allowing the functions to call the function directly in the instance of it if a functionFunc is not available.

Example
```
func (p *Plan) Path() motionplan.Path {
	if p.PathFunc == nil {
		return Plan.Path{}
	}
	return p.PathFunc()
}
```

Reply:
Functionality, I have never seen this be used but it is a pattern that appears in most testutils injects.

Reply:
This is old code I meant to take out since its no longer used.  Removing